lib/dns/resource.js

```diff
@@ -1125,6 +1125,8 @@ class Resource extends Struct {
             for (const rr of this.toGlue(types.SRV))
               res.additional.push(rr);
             key.signZSK(res.answer, types.SRV);
+            key.signZSK(res.additional, types.A);
+            key.signZSK(res.additional, types.AAAA);
```

Comment:
`key.signZSK` will throw an error if there are 2 RRs of the same type with a different name. Depending on `toSRVIP` and `toGlue`, I think this is possible. Removing DNSSEC from the `additional` section is one option, or handling that case in `key.signZSK` is another option. Down the codepath of `key.signZSK`, `bns.util.extractSet` is called without a name.

Reply:
Don't sign the additional section here

Reply:
> Don't sign the additional section here

This has been applied, if the additional section is signed then the response will be too large and fall back to TCP. The key is public anyways.

For more, see:
https://security.stackexchange.com/questions/91038/dnssec-why-are-rrsigs-returned-in-the-answer-not-additional-section